### PR TITLE
GyroSensor wait_until_angle_changed_by added direction_sensitive option

### PR DESCRIPTION
**Pull request by dwalton76**
For issue #534

### CODE REVIEW

**Comment by dwalton76** [comment]
Tested via
```
#!/usr/bin/env python3

from ev3dev2.sensor.lego import GyroSensor
from ev3dev2.auto import INPUT_1, INPUT_2

gyro = GyroSensor(INPUT_2)
print("{} angle {}".format(gyro, gyro.angle))
gyro.wait_until_angle_changed_by(-90, True)
print("{} angle {}".format(gyro, gyro.angle))
```

File: ev3dev2/sensor/lego.py [hunk 1]
```diff
@@ -626,16 +626,32 @@ def reset(self):
         self._ensure_mode(self.MODE_GYRO_ANG)
         self._direct = self.set_attr_raw(self._direct, 'direct', 17)
 
-    def wait_until_angle_changed_by(self, delta):
+    def wait_until_angle_changed_by(self, delta, honor_direction=False):
```
Comment (WasabiFan) [inline ev3dev2/sensor/lego.py]: I think "honor" might be confusing to non-native English speakers; could we call it something else? Maybe `direction_sensitive`? `specific_direction`?

File: ev3dev2/sensor/lego.py [hunk 2]
```diff
@@ -626,16 +626,32 @@ def reset(self):
...
         """
         assert self.mode in (self.MODE_GYRO_G_A, self.MODE_GYRO_ANG,
                              self.MODE_TILT_ANG),\
             'Gyro mode should be MODE_GYRO_ANG, MODE_GYRO_G_A or MODE_TILT_ANG'
         start_angle = self.value(0)
-        while abs(start_angle - self.value(0)) < delta:
-            time.sleep(0.01)
+
+        if honor_direction:
+            if delta > 0:
```
Comment (WasabiFan) [inline ev3dev2/sensor/lego.py:645]: This feels like a lot of duplicated code. Can we combine this to be one loop? I think this condition might work:

```python
while (
        (abs(self.value(0) - start_angle) < delta)  # the sensor still hasn't moved far enough... 
        or (honor_direction and copysign(delta, 1) == copysign(self.value(0) - start_angle, 1))) # … or, it has moved in the wrong direction
```


I'm not entirely sure my line break structure will fly with the Python interpreter, but you get the gist.

Comment (dwalton76) [reply ev3dev2/sensor/lego.py:645]: Personally I find the expanded/verbose model a little easier to follow :)

File: ev3dev2/sensor/lego.py (hunk 1 above)
Comment (dwalton76) [reply ev3dev2/sensor/lego.py]: ``direction_sensitive`` sounds good

File: ev3dev2/sensor/lego.py (hunk 2 above)
Comment (WasabiFan) [reply ev3dev2/sensor/lego.py:645]: I just want to remove the duplicated loop code... whether that's with a more clever loop condition or something else doesn't matter to me.

File: debian/changelog [hunk 3]
```diff
@@ -1,5 +1,6 @@
 python-ev3dev2 (2.0.0~beta3) stable; urgency=medium
 
+  * GyroSensor wait_until_angle_changed_by added honor_direction option
```
Comment (WasabiFan) [inline debian/changelog]: new parameter name

**Review by WasabiFan — APPROVED** [review]

**Pull request merged**